language/stdlib/modules/0L/Wallet.move

```diff
@@ -0,0 +1,411 @@
+        let f = borrow_global_mut<CommunityFreeze>(*&t.payer);
+        f.consecutive_rejections = f.consecutive_rejections + 1;
+        Vector::push_back(&mut c.rejected, t);
+        
+      };
+
+      i = i + 1;
+    };
+    
+    print(&0x01134);
+
+  }
+
+  // private function to tally vetos.
+  // checks if a voter is in the validator set.
+  // tallies everytime called. Only counts votes in the validator set.
+  // does not remove an address if not in the validator set, in case the validator returns
+  // to the set on the next tally.
+  fun tally_veto(index: u64): bool acquires CommunityTransfers {
+    let c = borrow_global_mut<CommunityTransfers>(0x0);
+    let t = Vector::borrow_mut<TimedTransfer>(&mut c.proposed, index);
+
+    let votes = 0;
+    let threshold = calculate_proportional_voting_threshold();
+    
+    let k = 0;
+    let len = Vector::length<address>(&t.veto.list);
+
+    while (k < len) {
+      let addr = *Vector::borrow<address>(&t.veto.list, k);
+      // ignore votes that are no longer in the validator set,
+      // BUT DON'T REMOVE, since they may rejoin the validator set, and shouldn't need to vote again.
+
+      if (LibraSystem::is_validator(addr)) {
+        votes = votes + NodeWeight::proof_of_weight(addr)
+      };
+      k = k + 1;
+    };
+
+    t.veto.count = votes;
+    t.veto.threshold = threshold;
+
+    return votes > threshold
+  }
+
+  // private function to get the total voting power of the validator set, and find the 2/3rds threshold
+  fun calculate_proportional_voting_threshold(): u64 {
+      let val_set_size = LibraSystem::validator_set_size();
+      let i = 0;
+      let voting_power = 0;
+      while (i < val_set_size) {
+        let addr = LibraSystem::get_ith_validator_address(i);        
+        voting_power = voting_power + NodeWeight::proof_of_weight(addr);
+        i = i + 1;
+      };
+      let threshold = voting_power * 2 / 3;
+      threshold
+  }
+
+  // Utility to list CommunityWallet transfers due, by epoch. Anyone can call this.
+  // This is used by VM in LibraAccount at epoch boundaries to process the wallet transfers.
+  public fun list_tx_by_epoch(epoch: u64): vector<TimedTransfer> acquires CommunityTransfers {
+      let c = borrow_global_mut<CommunityTransfers>(0x0);
+      // reset approved list
+      c.approved = Vector::empty<TimedTransfer>();
+      // loop proposed list
+      let pending = Vector::empty<TimedTransfer>();
+      let len = Vector::length(&c.proposed);
+      let i = 0;
+      while (i < len) {
+        let t = Vector::borrow(&c.proposed, i);
+        if (t.expire_epoch == epoch) {
+          
+          Vector::push_back<TimedTransfer>(&mut pending, *t);
+          // TODO: clear the freeze count on community wallet
+          // add to approved list
+        };
+        i = i + 1;
+      };
+      return pending
+    }
+    
+    public fun maybe_reset_rejection_counter(vm: &signer, wallet: address) acquires CommunityFreeze {
```

Comment:
Why maybe?

Reply:
Because there may not be any rejections to reset. Can change name if that's confusing.